reactive/swarm.py

```diff
@@ -0,0 +1,83 @@
+from charmhelpers.core import hookenv
+from charmhelpers.core import host
+from charmhelpers.core.templating import render
+from charms.reactive import when
+from charms.reactive import when_not
+from charms import reactive
+
+from dockeropts import DockerOpts
+
+from shlex import split
+from subprocess import check_call
+
+
+@when('etcd.available', 'docker.available')
+@when_not('swarm.available')
+def swarm_etcd_cluster_setup(etcd):
+    """
+    Expose the Docker TCP port, and begin swarm cluster configuration. Always
+    leading with the agent, connecting to the discovery service, then follow
+    up with the manager container on the leader node.
+    """
+    bind_docker_daemon()
+    start_swarm_etcd_agent(etcd.connection_string())
+    if hookenv.is_leader():
```

Comment:
Only the leader needs the etcd manager?  Do the other charms need to know etcd also?